### DISCUSSION TITLE
How can I use 'useStore' in router-guard

### DISCUSSION THREAD

**Discussion by Bacuuu** [vuejs/pinia#1216] [Help and Questions] [answered]
**what I want**
I use `Vue3.2.31`,`vue-router 4.0.13`,`pinia 2.0.13`. 
In Navigation Guards like  `router.foreach`  ,I want to change store state when the navigation changed.

**what is the problem**
I follow  this [doc](https://pinia.vuejs.org/core-concepts/outside-component-usage.html#single-page-applications).
error in console is 
> Uncaught Error: [🍍]: getActivePinia was called with no active Pinia. Did you forget to install pinia?

I have installed pinia,I don't how to resolve this problem.

**code is here!**
```ts
// main.ts
import router from './router'

app
.use(createPinia())
.use(router)
.mount('#app')
```

```ts
// router/index.ts
const router = createRouter({
  history: createWebHashHistory(),
  routes,
})
guard(router)
export default router
```

```ts
// function guard
export default function (router:Router) {
  const store = useDialogStore()
  // console.log(store)
  router.beforeEach((to, from) => {
    if (to.meta.requireAuth && !existToken()) {
      return {
        path: '/login',
      }
    }
  })
}
```

**Comment by posva** [comment, marked as answer]
Move the iseDialogStore within the navigation guard, before the if (to.meta.requiresAuth

**Reply by Bacuuu** [reply]
Now it works right!
I just reread the doc, there is an example about how to use store inside of a navigation guard with Vue Router. I'm so careless.
thanks!
